src/plugins/sqlalchemy_create_object_operator.py

```diff
@@ -1,68 +1,104 @@
+                as specified in the schema. Or it can be a compiled regular expression. The latter
+                is mostly useful if you want to create PostgreSQL objects for all tables in a
+                given schema. This is also the default value (all tables).
+            db_conn: DB connection URL.
+            ind_table: Whether to creates indices (as specified in the schema).
+            ind_extra_index: Whether to created additional indices.
+            *args:
+            **kwargs:
+        """
         super().__init__(*args, **kwargs)
         self.data_schema_name = data_schema_name
-        self.data_table_name = data_table_name
-        # Optionals
-        self.data_schema_env = f"{data_schema_env}." if data_schema_env else ""
+        if isinstance(data_table_name, str):
+            self.data_table_name = re.compile(data_table_name)
+        else:
+            self.data_table_name = data_table_name
+
         self.db_conn = db_conn
         self.ind_table = ind_table
         self.ind_extra_index = ind_extra_index
 
     def execute(self, context=None):
-        """Executes the 'generate_db_object' method from schema-tools.
-        Which leads to the creation of tables and/or an index on the identifier (as specified in the data JSON schema).
-        By default both tables and the identifier and 'many-to-many table' indexes are created.
-        By setting the boolean indicators in the method parameters, tables or an identifier index (per table) can be created.
+        """Executes the ``generate_db_object`` method from schema-tools.
+
+        Which leads to the creation of tables and/or an index on the identifier (as specified in
+        the data JSON schema). By default both tables and the identifier and 'many-to-many
+        table' indexes are created. By setting the boolean indicators in the method parameters,
+        tables or an identifier index (per table) can be created.
         """
-        data_schema_url = f"{SCHEMA_URL.split('//')[0]}//{self.data_schema_env}{SCHEMA_URL.split('//')[1]}{self.data_schema_name}/{self.data_schema_name}"
+        data_schema_url = "/".join(
+            map(lambda s: s.strip("/"), (SCHEMA_URL, self.data_schema_name, self.data_schema_name))
+        )
         data = schema_fetch_url_file(data_schema_url)
         engine = _get_engine(self.db_conn)
         parent_schema = SchemaType(data)
         dataset_schema = DatasetSchema(parent_schema)
         importer = BaseImporter(dataset_schema, engine)
+        self.log.info(
+            "data_schema_url='%s', engine='%s', ind_table='%s', ind_extra_index='%s'.",
+            data_schema_url,
+            engine,
+            self.ind_table,
+            self.ind_extra_index,
+        )
+
         for table in data["tables"]:
-            if (
-                self.data_schema_name + "_" + table["id"]
-                == f"{self.data_table_name}"
-            ):
+            self.log.info("Considering table '%s'.", table["id"])
```

Comment:
Maybe add extra precaution, because table-ids can be in camelCase in the schema (although it is rare).

Reply:
Hmm, de aanname hier, en zoals gedocumenteerd bij de `__init__` parameter `data_table_name`, is dat het hier gaat om de tabel naam _as specified in the schema_. Dus als je camelCase in het schema gebruikt, dan moet je dat hier ook doen. Met dien verstande dat de dataset naam er ook nog eens als prefix bij werd geplakt. Het was bestaande code waarop ik voortborduurde. Maar heeft me wel even bezig gehouden. Had uiteindelijk besloten het zo te laten, maar ...

We moeten eigenlijk afspreken wat het _canonical format_  voor naamgeving is. Het Amsterdam Schema heeft er eentje specifiek voor de schemas. In PostgreSQL is het het allemaal snake_case. Maar in onze code hebben we de neiging steeds heen en weer te converteren. Conversie zouden alleen moeten gebeuren op de _boundaries_.

Hier is geen sprake van een _boundary_. Het schema is reeds geparseerd (=boundary). Dit is ook geen code die daadwerkelijk tabellen aanmaakt (=boundary). Dit is code die daar tussenin zit. Hier wil ik aan tabelnamen en andere entiteitnamen kunnen refereren in een _canonical format_, en dus niet hoeven nadenken of ik nu te maken heb met camelCase of snake_case.

Daar we allemaal Python programmeur zijn zou ik willen voorstellen dat snake_case ons _canonical format_ is. Dat zou betekenen dat zodra een schema geparseerd is in een reeks van Python datastructuren, de naamgeving in die datastructuren allemaal in snake_case is. Dat is momenteel niet het geval. Lijkt jou dat ook een wenselijke situatie?

Reply:
Helemaal eens met wat je zegt over boundaries. Doet me denken aan de tijd dat unicode werd geintroduceerd in python, dat gaf toen ook veel verwarring op het moment dat er teveel interne conversies plaatsvonden. Daar was die boundaries-regel toen ook erg nuttig.

Vwb. het canonical formaat, dat lijkt mij wel een goeie aanpak, en idd, voor python zou dan de snake_casing het meest natuurlijk aanvoelen. De "inward" boundary zit dan vooral in `schematools.types`.